libusb/libusb.h

```diff
@@ -1588,6 +1588,10 @@ int LIBUSB_CALL libusb_get_configuration(libusb_device_handle *dev,
 	int *config);
 int LIBUSB_CALL libusb_get_device_descriptor(libusb_device *dev,
 	struct libusb_device_descriptor *desc);
+int LIBUSB_CALL libusb_get_serial_string_descriptor(libusb_device *dev,
+	unsigned char *data, int length);
+int LIBUSB_CALL libusb_get_serial_string_descriptor_ascii(libusb_device *dev,
+	unsigned char *data, int length);
```

Comment:
as in the previous PR, I don't thing we should limit this functionality to serial number only
we probably should extend this to product and manufacturer as well
and I think would be great to have a single function for all three of them

@mcuee, @tormodvolden any ideas on the API?
the most obvious that comes to my mind is to specify a new "enum" for the string kind, but I'm not sure if that's the best way to do so

Reply:
I'd say the one significant difference is that each different descriptor function returns a different descriptor, which is a _different struct/type_. All potential new functions will return the same thing - a string.

Reply:
Good point.